frontend/document/credocument.lua

```diff
@@ -1745,8 +1756,14 @@ function CreDocument:setupCallCache()
     -- Dump statistics on close
     if do_stats then
         self.close = function(_self)
-            CreDocument.close(_self)
             dumpStats()
+            self._callCacheReset()
```

Comment:
Is that always required with `ffi/lru.lua` ?

Reply:
Not specific to it, no, I just feel better explicitly killing the reference tree *now*, because otherwise it's possibly alive until the next CreDocument.

Reply:
Okay, I went down a fairly hellish rabbit hole, but the TL;DR is:

* Explicitly `nil`'ing the caches is *necessary* to get them collected by the GC.

(So, slightly different approach this this, but haven'"t pushed yet. Also, fixed a minor issue with :clear in lru that migh have prevented collection of the final item pop'ed).

That would seem to imply that we're keeping a reachable reference to Document instances.... somehow, somewhere.

But after a few hours of stupid experiment, I hit a wall trying to unwind that one, so :/. (AFAICT, we do nil *direct* references to Document instances, but we may have some alive to ReaderUI/FileManager, and taking care of those might be tricky...).

Anyway, if you want to go mad, these tools are fun/scary: https://github.com/alexbfreeman/lua-traverse & https://code.matthewwild.co.uk/luatraverse

(They don't appear to deal with cdata well/at all, so, I just ignore them in the final gc.edge function).